editor/src/clj/editor/gl/shader.clj

```diff
@@ -634,21 +654,21 @@ of GLSL strings and returns an object that satisfies GlBind and GlEnable."
 
                 sampler-name->uniform-names
                 (into array-sampler-name->uniform-names
-                      (keep (fn [[uniform-name uniform-info]]
+                      (keep (fn [uniform-info]
                               (when (sampler-uniform-type? (:type uniform-info))
-                                (when-not (array-sampler-uniform-name? uniform-name)
-                                  (pair uniform-name [uniform-name])))))
+                                (when-not (array-sampler-uniform-name? (:name uniform-info))
+                                  (pair (:name uniform-info) [(:name uniform-info)])))))
                       uniform-infos)
 
                 sampler-index->sampler-name
                 (->> sampler-name->uniform-names
                      (mapv (fn [[sampler-name uniform-names]]
                              (pair sampler-name
-                                   (uniform-infos (first uniform-names)))))
+                                   (uniform-name->uniform-infos (first uniform-names) uniform-infos))))
                      (sort-by (comp :index second))
                      (mapv first))]
             {:program program
-             :uniform-infos uniform-infos
+             :uniform-infos (strip-resource-namespaces uniform-infos resource-binding-namespaces)
```

Comment:
Does this really need to happen in two phases?

It seems we could do this when we're constructing `uniform-infos` above?